### PR TITLE
Cleanup learning persistence

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs [hunk 1]
```diff
@@ -83,16 +91,14 @@ public object Read()
             return serializer.Deserialize(jsonReader, manifest.SagaEntityType);
         }
 
-        static void WriteWithinLock(Action<JsonWriter, IContainSagaData> action, JsonWriter writer, string targetPath, DateTime lastModificationSeenAt, IContainSagaData sagaData = null)
+        static void WriteWithinLock(Action action, string targetPath, DateTime lastModificationSeenAt)
```
Comment (bording) [inline src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs]: 👎 To this change. The way it was before, we completely avoid having to close over any values, which should prevent the need for a per-call delegate allocation.

Comment (SimonCropp) [inline src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs]: for a dev persistence?

**Comment by bording** [comment]
@SimonCropp Looking at `SagaStorageFile`, one additional cleanup I do see is that we should be able to stop storing `fileStream` and instead just store the file name.

File: src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs (hunk 1 above)
Comment (bording) [inline src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs]: If the code is already there, why change it?

Comment (bording) [inline src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs]: I guess let's see what the other maintainers think.

**Comment by SimonCropp** [comment]
ok. will close this and just remove the TODO

**Pull request closed without merging**